crates/solver/src/settlement_submission.rs

```diff
@@ -127,15 +146,63 @@ impl SolutionSubmitter {
                         *flashbots_tip,
                     )
                     .await;
-                match result {
-                    Ok(Some(hash)) => Ok(hash),
-                    Ok(None) => bail!("transaction did not get mined in time"),
-                    Err(err) => Err(err),
-                }
+                process_result(result, "flashbots", metrics)
             }
             TransactionStrategy::DryRun => {
-                dry_run::log_settlement(account, &self.contract, settlement).await
+                let result = dry_run::log_settlement(account, &self.contract, settlement).await;
+                process_simple_result(result, "dry_run", metrics)
             }
         }
     }
 }
+
+#[derive(prometheus_metric_storage::MetricStorage, Clone, Debug)]
+#[metric(subsystem = "settlement_submission")]
+struct SettlementSubmissionMetrics {
+    /// Number of completed API requests.
+    #[metric(labels("transaction_strategy", "status"))]
+    requests_complete: prometheus::CounterVec,
+}
+
+fn process_result(
+    result: Result<Option<TransactionReceipt>>,
+    transaction_strategy: &str,
+    metrics: &SettlementSubmissionMetrics,
+) -> Result<TransactionReceipt> {
+    match result {
+        Ok(Some(hash)) => {
+            metrics
+                .requests_complete
+                .with_label_values(&[transaction_strategy, "success"])
+                .inc();
+            Ok(hash)
+        }
+        Ok(None) => {
+            metrics
+                .requests_complete
+                .with_label_values(&[transaction_strategy, "timeout"])
+                .inc();
+            Err(TransactionTimeoutError.into())
+        }
+        Err(err) => {
+            metrics
+                .requests_complete
+                .with_label_values(&[transaction_strategy, "error"])
+                .inc();
+            Err(err)
+        }
+    }
+}
+
+fn process_simple_result(
```

Comment:
It's just an internal function, but I generally don't like naming things "simple" as it doesn't really tell you anything about what this function does (process_tx_hash vs. process_optional_tx_hash might be better). E.g.  it turns out that `NaiveSolver` wasn't a great name as it's not really explanatory.

Here specifically though, can't we just call `process_result(result.map(Some), transaction_strategy, metrics)`? This way we don't have to duplicate label names and other shared logic.